python/tank/templatekey.py

```diff
@@ -292,31 +304,164 @@ def __init__(self,
+            return
+
+        if isinstance(format_spec, basestring) is False:
+            msg = "format_spec for IntegerKey %s is not of type string: %s"
             raise TankError(msg % (name, str(format_spec)))
 
+        if len(format_spec) == 0:
+            raise TankError("format_spec can't be empty.")
+
+        matches = self._FORMAT_SPEC_RE.match(format_spec)
+        if not matches:
+            raise TankError("format_spec for IntegerKey %s has to either be a number (e.g. '3') or "
+                            "a 0 followed by a number (e.g. '03'), not '%s'" % (name, format_spec))
+
+        groups = matches.groups()
+        # groups[0] is either '' or '0', in which case the padding is ' '
+        self._padding_char = groups[0] or ' '
+        # groups[1] is the minimum width of the number.
+        self._minimum_width = int(groups[1])
         self.format_spec = format_spec
 
+    def _init_strict_matching(self, name, strict_matching):
+        """
+        Asserts that the strict_matching parameter is a valid value.
+
+        :param name: Name of this template key.
+        :param strict_matching: Parameter to be validated.
+
+        :raises TankError: Raised when the parameter is not a boolean.
+        """
+        # make sure that strict_matching is not set or that it is a boolean
+        if not(strict_matching is None or isinstance(strict_matching, bool)):
+            msg = "strict_matching for IntegerKey %s is not of type boolean: %s"
+            raise TankError(msg % (name, str(strict_matching)))
+
+        # If there is a format and strict_matching is set, there's an error, since there
+        # is no format to enforce or not.
+        if self.format_spec is None and strict_matching is not None:
+            raise TankError("strict_matching can't be set if there is no format_spec")
+
+        # By default, if strict_matching is not set but there is a format spec, we'll
+        # strictly match.
+        if strict_matching is None and self.format_spec is not None:
+            strict_matching = True
+
+        if strict_matching:
+            # This regular expression is blind to the actual length of the string for performance
+            # reasons. Code that uses it should test that the string's length is of
+            # self._minimum_width first. It first matches up to n-1 0s. It then matches either a
+            # single 0, or an actual two digit number that doesn't start with 0.
+            self._strict_validation_re = re.compile("^%s{0,%d}((%s)|0)$" % (
+                self._padding_char, self._minimum_width - 1, self._NON_ZERO_POSITIVE_INTEGER_EXP)
+            )
+        else:
+            self._strict_validation_re = None
+
+        self.strict_matching = strict_matching
+
     def validate(self, value):
+        """
+        Validates the value.
+
+        :param value: Value to validate.
 
+        :returns: True is the validation was succesfull, False otherwise.
+        """
```

Comment:
wondering if there could be a single return from this method? Does that make sense?